lib/ibm_cloud_sdk_core/jwt_token_manager.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require("http")
+require("json")
+require("jwt")
+require("rbconfig")
+require_relative("./version.rb")
+
+TOKEN_NAME = "access_token"
+
+module IBMCloudSdkCore
+  # Class to manage JWT Token Authentication
+  class JWTTokenManager
+    def initialize(vars)
+      defaults = {
+        token_info: nil,
+        url: nil,
+        access_token: nil
+      }
+      vars = defaults.merge(vars)
+
+      @url = vars[:url]
+      @token_info = vars[:token_info]
+      @user_access_token = vars[:access_token]
+      @time_to_live = nil
+      @expire_time = nil
+      @disable_ssl_verification = false
+    end
+
+    def token
+      if !@user_access_token.nil?
+        @user_access_token
+      elsif !@token_info.nil?
+        @token_info[TOKEN_NAME]
+      elsif @token_info.nil? || token_expired?
```

Comment:
Minor change needed here - need to check if the token is expired before returning the saved token (if we have a token and its expired, we will never make it to the last branch and the request will fail).

The last two branches need to be switched. Something like this:
```ruby
    def token
      if !@user_access_token.nil?
        @user_access_token
      elsif @token_info.nil? || token_expired?
        token_info = request_token
        save_token_info(token_info: token_info)
        @token_info[TOKEN_NAME]
      else
        @token_info[TOKEN_NAME]
      end
    end
```

Reply:
Sounds good. If I do that I need to check for `@token_info.nil? || (!@token_info.nil? && token_expired?)` because if the token is provided during initialize we don't save expiry time and time to live, we just save `@user_access_token`. I either need to save the whole thing in the constructor or update the logic. What do you think?

Reply:
nvm, if the user provides the token we don't store the token, and if we create token for them then we have `token_info`. I don't need that extra check in logic.

Reply:
updated.